pkg/safeguards/preprocessing/preprocessing.go

```diff
@@ -16,6 +17,41 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/filesys"
 )
 
+// Given a path, will determine if it's Kustomize, Helm, a directory of manifests, or a single manifest
+func GetManifestFiles(manifestsPath string) ([]safeguards.ManifestFile, error) {
```

Comment:
Where is this  function called? You have some repeated logic here and in GetManifestFilesFromDir

Reply:
getManifestFilesFromDir is used to recursively process a directory of .yaml files. And GetManfiestFiles() calls this func, in addition to helm, kustomize, and a single .yaml file provided as the path, so I left it as is since RenderHelm and RenderKustomize functions handle processing directories differently. If this is too much overlap, we could look into refactoring all functions in anther PR. However, at the moment, I don't see how we could reduce code between these two functions.